pkg/topology/kademlia/kademlia.go

```diff
@@ -1493,10 +1545,6 @@ func (k *Kad) IsBalanced(bin uint8) bool {
 	k.depthMu.RLock()
 	defer k.depthMu.RUnlock()
 
-	if int(bin) > len(k.commonBinPrefixes) {
-		return false
-	}
-
```

Comment:
this shouldn't be removed, but condition should be changed to:

`int(bin) >= len(k.commonBinPrefixes)`